presto-main/src/main/java/com/facebook/presto/operator/OrderByOperator.java

```diff
@@ -218,4 +295,60 @@ public void close()
         }
         return types.build();
     }
+
+    public WorkProcessor<Page> unspill()
+    {
+        List<WorkProcessor<Page>> spilledPages = getSpilledPages();
+
+        if (spilledPages.isEmpty()) {
+            pageIndex.sort(sortChannels, sortOrder);
+
+            return WorkProcessor.fromIterator(pageIndex.getSortedPages());
+        }
+
+        return mergeSpilledPagesAndCurrentPagesIndex(pageIndex, spilledPages);
+    }
+
+    public List<WorkProcessor<Page>> getSpilledPages()
+    {
+        if (!spiller.isPresent()) {
+            return ImmutableList.of();
+        }
+
+        return spiller.get().getSpills().stream()
+                .map(WorkProcessor::fromIterator)
+                .collect(Collectors.toList());
+    }
+
+    public WorkProcessor<Page> mergeSpilledPagesAndCurrentPagesIndex(PagesIndex currentPagesIndex, List<WorkProcessor<Page>> spilledPages)
+    {
+        PageWithPositionComparator pageWithPositionComparator = new SimplePageWithPositionComparator(sourceTypes, sortChannels, sortOrder);
+        WorkProcessor<Page> pageIndexPages = WorkProcessor.fromIterator(currentPagesIndex.getSortedPages());
```

Comment:
Please shortcircut the case when spill did not happen. In such case we don't need to merge pages index rows, but can use `PagesIndex#getSortedPages` directly. That should be faster

Reply:
That is what happens today in ``unspill``. We will return an ``iterator`` of the pages in the pagesIndex, by calling ``getSortedPages`` directly